### PR TITLE
Jk/fix/cleancode

### PR DESCRIPTION
**Pull request by johnnykeats**
Clean.
Roles, resources, documentation, etc.

### CODE REVIEW

File: roles/core/conman/handlers/main.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-- name: Restart conman
+- name: service █ Restart conman
```
Comment (actatux) [inline roles/core/conman/handlers/main.yml:2]: You have to update the `notify:` in tasks.

**Comment by johnnykeats** [comment]
Indeed. Fixed.

**Comment by actatux** [comment]
Hi @johnnykeats ,
I just noticed repositories_client is still using historical facts. (`ansible_*` instead of `ansible_facts.*`, see [Ansible fact namespacing](https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.5.html#ansible-fact-namespacing)). After a search, I found some other facts to update.

Since you are cleaning the code, can you add some commits to your branch to fix this? (I could push to the branch but don't want to mess with your WIP)

```
# grep -r ' ansible_' roles/|grep -vE '(ansible_facts|ansible_managed|ansible_skip_tags)'
roles/core/bluebanquise/molecule/default/verify.yml:      that: ansible_all_ipv4_addresses | ipaddr
roles/core/bluebanquise/molecule/default/verify.yml:    command: ansible localhost -c local -m assert -a "that={{ ansible_all_ipv4_addresses }}|ipaddr"
roles/core/repositories_client/tasks/main.yml:  include_tasks: "{{ (ansible_distribution|lower|replace(' ','_')) }}_{{ ansible_distribution_major_version }}.yml"
roles/core/time/molecule/default/verify.yml:    - name: Ansible fact - ansible_date_time.iso8601
roles/core/time/molecule/default/verify.yml:        msg: "Timestamp format: {{ ansible_date_time }}"
roles/core/time/molecule/default/verify.yml:        that: "'CEST' in ansible_date_time.tz"
```

- ansible_all_ipv4_addresses -> ansible_facts.all_ipv4_addresses
- ansible_distribution -> ansible_facts.distribution
- ansible_distribution_major_version -> ansible_facts.distribution_major_version
- ansible_date_time.iso8601 -> ansible_facts.date_time.iso8601
- ansible_date_time -> ansible_facts.date_time
- ansible_date_time.tz -> ansible_facts.date_time.tz

Thanks!

File: roles/core/hosts_file/templates/hosts.j2 [hunk 2]
```diff
@@ -9,71 +9,71 @@
...
     {% endif %}
-    {% if host_dict['bmc']['alias'] is defined and not none and host_iceberg == j2_current_iceberg %}
+    {% if host_dict['bmc']['alias'] is defined and host_dict['bmc']['alias'] is not none and host_iceberg == j2_current_iceberg %}
       {% for alias in host_dict['bmc']['alias'] %}{{ bmc_alias_list.append(alias) }}{% endfor %}
     {% endif %}
 
-    {%- if (host_dict['bmc'].name is defined and not none) and (host_dict['bmc'].ip4 is defined and not none) %}
- {{host_dict['bmc'].ip4}} {{host_dict['bmc'].name}} {{bmc_alias_list|join(' ')}}
+    {%- if (host_dict['bmc'].name is defined and host_dict['bmc'].name is not none) and (host_dict['bmc'].ip4 is defined and host_dict['bmc'].ip4 is not none) %}
+ {{ host_dict['bmc'].ip4 }} {{ host_dict['bmc'].name }} {{ bmc_alias_list|join(' ') }}
```
Comment (actatux) [inline roles/core/hosts_file/templates/hosts.j2:56]: Can you remove the leading whitespace for bmc?

```suggestion
{{ host_dict['bmc'].ip4 }} {{ host_dict['bmc'].name }} {{ bmc_alias_list|join(' ') }}
```

It will produce:
```
TASK [hosts_file : Template >> /etc/hosts] ***************************
--- before: /etc/hosts
+++ after: /root/.ansible/tmp/ansible-local-290445fgv3af7/tmpor5ujxlc/hosts.j2
@@ -11,7 +11,7 @@

 10.10.0.1 management1 management1.tumulus.tld management1-ice1-1
 10.20.0.1 management1-interconnect-1
- 10.10.100.1 bmanagement1 
+10.10.100.1 bmanagement1 

 10.10.3.1 c001 c001.tumulus.tld c001-ice1-1
 10.20.3.1 c001-interconnect-1   
@@ -27,7 +27,7 @@

 10.10.2.1 login1 login1.tumulus.tld login1-ice1-1
 10.20.2.1 login1-interconnect-1 
- 10.10.102.1 blogin1 
+10.10.102.1 blogin1 


```

Comment (johnnykeats) [reply roles/core/hosts_file/templates/hosts.j2:56]: Will do both.

**Comment by johnnykeats** [comment]
First pass done. Enough for this PR.
Need then to clean documentation, and maybe addons + advanced_core in next step.

File: roles/core/bluebanquise/tasks/main.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ---
-- name: Gathering OS specific variables
+- name: include_vars ░ Gather OS specific variables
```
Comment (actatux) [inline roles/core/bluebanquise/tasks/main.yml:2]: Is it expected to have '░' here while package and service tasks use '█' separator?

File: roles/core/nfs_client/readme.rst [hunk 4]
```diff
@@ -4,21 +4,53 @@ NFS client
...
 Instructions
 ^^^^^^^^^^^^
 
 All configuration is done in *group_vars/all/general_settings/nfs.yml*.
 
-See the **nfs_server** role isntructions for more details.
+See the **nfs_server** role instructions for more details.
 
-Simply note that path that should be mounted will be automatically created by this role if they do not exist.
+Simply note that path that should be mounted will be automatically created by
```
Comment (actatux) [inline roles/core/nfs_client/readme.rst:17]:
```suggestion
Simply note that paths that should be mounted will be automatically created by
```

File: roles/core/nfs_client/readme.rst [hunk 5]
```diff
@@ -4,21 +4,53 @@ NFS client
...
+
+Input
+^^^^^
+
+Mandatory inventory vars:
+
+**hostvars[inventory_hostname]**
+
+* nfs[item]
+  * .mount
```
Comment (actatux) [inline roles/core/nfs_client/readme.rst:28]: The RST rendering in GitHub is weird: https://github.com/bluebanquise/bluebanquise/blob/16308753e084a3920c8aa3666a5ebe053d3dcaee/roles/core/nfs_client/readme.rst

Are you sure about the syntax?

File: roles/core/nfs_server/handlers/main.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ---
```
Comment (actatux) [inline roles/core/nfs_server/handlers/main.yml:2]: blank line

File: roles/core/pxe_stack/handlers/main.yml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 ---
```
Comment (actatux) [inline roles/core/pxe_stack/handlers/main.yml:2]: blank line

File: roles/core/pxe_stack/readme.rst [hunk 8]
```diff
@@ -20,7 +19,8 @@ ip address and filename to use.
 **Files location**
 """"""""""""""""""
 
-* PXE boot files are located in /var/www/html/preboot_execution_environment/.
+* PXE boot files are located in {/var/www/html|/srv/www/htdocs}/preboot_execution_environment/,
```
Comment (actatux) [inline roles/core/pxe_stack/readme.rst:22]: I don't see any way to use a path different than /var/www/html/ in the role yet.
Might be too early to document alternative OS?

File: roles/core/pxe_stack/tasks/CentOS_7.yml [hunk 9]
```diff
@@ -1,10 +1,9 @@
 ## Tested on Centos 7.6
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/CentOS_7.yml:2]: blank line

File: roles/core/pxe_stack/tasks/CentOS_8.yml [hunk 10]
```diff
@@ -1,10 +1,9 @@
 ## Tested on RedHat 8.0
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/CentOS_8.yml:2]: blank line

File: roles/core/pxe_stack/tasks/CentOS_7.yml [hunk 11]
```diff
@@ -1,10 +1,9 @@
...
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/CentOS_7.yml:1]: miss `---` ?

File: roles/core/pxe_stack/tasks/CentOS_8.yml [hunk 12]
```diff
@@ -1,10 +1,9 @@
...
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/CentOS_8.yml:1]: miss `---` ?

File: roles/core/pxe_stack/tasks/RedHat_7.yml [hunk 13]
```diff
@@ -1,10 +1,9 @@
 ## Tested on Centos 7.6
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/RedHat_7.yml:1]: miss `---` ?

File: roles/core/pxe_stack/tasks/RedHat_7.yml [hunk 14]
```diff
@@ -1,10 +1,9 @@
...
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/RedHat_7.yml:2]: blank line

File: roles/core/pxe_stack/tasks/RedHat_8.yml [hunk 15]
```diff
@@ -1,10 +1,9 @@
 ## Tested on RedHat 8.0
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/RedHat_8.yml:1]: miss `---` ?

File: roles/core/pxe_stack/tasks/RedHat_8.yml [hunk 16]
```diff
@@ -1,10 +1,9 @@
...
```
Comment (actatux) [inline roles/core/pxe_stack/tasks/RedHat_8.yml:2]: blank line

File: roles/core/pxe_stack/templates/menu.ipxe.j2 [hunk 17]
```diff
@@ -103,13 +103,10 @@ iseq ${platform} efi && goto bootdisk-efi || goto bootdisk-legacy
...
-:memtest
-echo Booting Memtest86+
-boot http://${next-server}/preboot_execution_environment/bin/memtest86+-5.01.bin || shell
-
-:minimal_live_linux
-echo Booting Minimal Live Linux
-chain http://${next-server}/preboot_execution_environment/equipment_ipxe_configurations/minimal.ipxe
-
-
+#:memtest
```
Comment (actatux) [inline roles/core/pxe_stack/templates/menu.ipxe.j2:106]: Why do you drop this?

File: roles/core/pxe_stack/templates/pxe_parameters.yml.j2 [hunk 18]
```diff
@@ -1,4 +1,6 @@
 pxe_parameters:
+  http_html_path:
+  http_cgi_path:
```
Comment (actatux) [inline roles/core/pxe_stack/templates/pxe_parameters.yml.j2:3]: Unused variables? Might be better to introduce these in a different PR, with the related code?

File: roles/core/repositories_client/handlers/main.yml [hunk 19]
```diff
@@ -1,5 +1,5 @@
 # Handler to clean yum metadata cache
```
Comment (actatux) [inline roles/core/repositories_client/handlers/main.yml:1]: miss `---` ?

File: roles/core/repositories_client/readme.rst [hunk 20]
```diff
@@ -6,6 +6,43 @@ Description
...
+    - myrepo
+
+Advanced one, to be combined with simple one as desired:
+
+.. code-block:: yaml
+
+  repositories:
+    - os
+    - bluebanquise
+    - name: epel
```
Comment (actatux) [inline roles/core/repositories_client/readme.rst:33]: While technically functional, @oxedions wanted to recommend the use of `external_repositories` for repositories hosted outside of the cluster.
No problem for me to deprecate `external_repositories` though.

File: roles/core/repositories_client/tasks/redhat_7.yml [hunk 21]
```diff
@@ -1,10 +1,10 @@
 ---
```
Comment (actatux) [inline roles/core/repositories_client/tasks/redhat_7.yml:2]: blank line

File: roles/core/repositories_server/tasks/main.yml [hunk 22]
```diff
@@ -33,14 +35,14 @@
   tags:
     - firewall
 
-- name: Install packages
+- name: "package █ Install {{ repositories_server_packages_to_install }} packages"
```
Comment (actatux) [inline roles/core/repositories_server/tasks/main.yml:38]: Add pretty printing of list with join?

File: roles/core/repositories_server/tasks/main.yml [hunk 23]
```diff
@@ -33,14 +35,14 @@
...
   package:
     name: "{{ repositories_server_packages_to_install }}"
     state: present
   tags:
     - package
 
-- name: Manage services state
+- name: "service █ Manage {{ repositories_server_services_to_start }} services state"
```
Comment (actatux) [inline roles/core/repositories_server/tasks/main.yml:45]: Add pretty printing of list with join?

File: roles/core/repositories_server/vars/main.yml [hunk 24]
```diff
@@ -1,2 +1,3 @@
 ---
-role_version: 1.0.2
+
```
Comment (actatux) [inline roles/core/repositories_server/vars/main.yml:2]: blank line

File: roles/core/set_hostname/vars/main.yml [hunk 25]
```diff
@@ -1 +1,3 @@
-role_version: 1.0.1
+---
+
```
Comment (actatux) [inline roles/core/set_hostname/vars/main.yml:2]: blank line

File: roles/core/ssh_master/readme.rst [hunk 26]
```diff
@@ -4,7 +4,7 @@ SSH master
 Description
 ^^^^^^^^^^^
 
-This role configure the ssh access of know hosts to make this access through nodes main network.
+This role configure the ssh access of inventory know hosts to ensure ssh access through nodes main network.
```
Comment (actatux) [inline roles/core/ssh_master/readme.rst:7]:
```suggestion
This role configures the ssh access of inventory known hosts to ensure ssh access through nodes main network.
```

File: roles/core/ssh_master/readme.rst [hunk 27]
```diff
@@ -84,15 +93,28 @@ first one to break in case of uncoherent inventory. If this happens, check your
...
 
+Input
+^^^^^
+
+Mandatory inventory vars:
+
+**hostvars[hosts]**
+
+* network_interfaces[item]
+
```
Comment (actatux) [inline roles/core/ssh_master/readme.rst:104]: \+ optional var `security.ssh.hostkey_checking` for hostvars[inventory_host] ?

File: roles/core/ssh_master/vars/main.yml [hunk 28]
```diff
@@ -1 +1,3 @@
-role_version: 1.0.1
+---
+
```
Comment (actatux) [inline roles/core/ssh_master/vars/main.yml:2]: blank line

File: roles/core/ssh_slave/readme.rst [hunk 29]
```diff
@@ -13,11 +13,25 @@ This role will ensure remote hosts is having currently defined ssh authorized pu
 
 These keys are set in file */etc/bluebanquise/inventory/group_vars/all/all_equipments/authentication.yml*.
 
-Keep in mind that this file can be precedenced with equipment_profiles groups or iceberg groups.
+Keep in mind that this file can be precedenced with equipment_profiles groups.
```
Comment (actatux) [inline roles/core/ssh_slave/readme.rst:16]:
```suggestion
Keep in mind that this file can be preceded with equipment_profiles groups.
```

File: roles/core/ssh_slave/readme.rst [hunk 30]
```diff
@@ -13,11 +13,25 @@ This role will ensure remote hosts is having currently defined ssh authorized pu
 
 These keys are set in file */etc/bluebanquise/inventory/group_vars/all/all_equipments/authentication.yml*.
```
Comment (actatux) [inline roles/core/ssh_slave/readme.rst:14]: Use relative path ?

File: roles/core/ssh_slave/tasks/main.yml [hunk 31]
```diff
@@ -1,11 +1,9 @@
 ---
```
Comment (actatux) [inline roles/core/ssh_slave/tasks/main.yml:2]: blank line

File: roles/core/ssh_slave/vars/main.yml [hunk 32]
```diff
@@ -1 +1,3 @@
-role_version: 1.0.1
+---
+
```
Comment (actatux) [inline roles/core/ssh_slave/vars/main.yml:2]: blank line

File: roles/core/time/handlers/main.yml [hunk 33]
```diff
@@ -1,6 +1,6 @@
 ---
```
Comment (actatux) [inline roles/core/time/handlers/main.yml:2]: blank line

File: roles/core/time/molecule/default/verify.yml [hunk 34]
```diff
@@ -9,13 +9,13 @@
       assert:
         that: "'chrony' in ansible_facts.packages"
 
-    - name: Ansible fact - ansible_date_time.iso8601
+    - name: Ansible fact - ansiblefacts._date_time.iso8601
```
Comment (actatux) [inline roles/core/time/molecule/default/verify.yml:12]: typo
```suggestion
    - name: Ansible fact - ansible_facts.date_time.iso8601
```

File: roles/core/time/tasks/main.yml [hunk 35]
```diff
@@ -40,24 +42,24 @@
...
 
-- name: Set timezone
+- name: "timezone █ Set system to {{ time_zone }} timezone"
   timezone:
     name: "{{ time_zone }}"
 
 - meta: flush_handlers
 
-- name: Manage services state
+- name: "service █ Manage {{ time_services_to_start }} services state"
```
Comment (actatux) [inline roles/core/time/tasks/main.yml:62]: Add pretty print of list with join?

File: roles/core/time/vars/main.yml [hunk 36]
```diff
@@ -1,2 +1,3 @@
 ---
-role_version: 1.0.2
+
```
Comment (actatux) [inline roles/core/time/vars/main.yml:2]: blank line

File: roles/core/bluebanquise/tasks/main.yml (hunk 3 above)
Comment (actatux) [reply roles/core/bluebanquise/tasks/main.yml:2]: I've noticed afterward this is for all the tasks with tag internal, so seems expected.

File: roles/core/pxe_stack/templates/menu.ipxe.j2 (hunk 17 above)
Comment (johnnykeats) [reply roles/core/pxe_stack/templates/menu.ipxe.j2:106]: Doesnt work right now. Should be reworked. So commented for now.

File: roles/core/repositories_client/readme.rst (hunk 20 above)
Comment (johnnykeats) [reply roles/core/repositories_client/readme.rst:33]: Ox is right, updating.

File: roles/core/nfs_client/readme.rst (hunk 5 above)
Comment (johnnykeats) [reply roles/core/nfs_client/readme.rst:28]: Missing one space.

File: roles/core/repositories_client/readme.rst (hunk 20 above)
Comment (johnnykeats) [reply roles/core/repositories_client/readme.rst:33]: Assuming we can deprecate external_repositories, makes sense here. Adding a task in the list.

**Comment by johnnykeats** [comment]
Impressive review.
Updated, also corrected few other things.
